### PR TITLE
Show notice to admin if does not opted in for "Usage Tracking" 

### LINKED ISSUE #5449: Admin should have admin setting to opt in and opt out from usage tracking
## User Story

As a admin, I want to control `Usage Tracking` feature so that I can enable and disable it.

## Details
In upcoming plugin major version , We are adding usage tracking feature. We will only non-sensitive data about how GiveWP is setup and managed. We do not track or store personal data of plugin user. Admin can turn off tracking with admin setting.

## Acceptance Criteria

-   [x] Admin can see setting `Usage Tracking` from `Settings->Advanced`.
-   [x] Admin see notice nag if `Usage Tracking` disabled.
-   [x] Each admin user can disabled notice nag for one day.

### PR DESCRIPTION
**Pull request by ravinderk**
Resolves #5449 

## Description

I added an admin setting to onboard admin and a notice nag to encourage admin to opted in the website for usage tracking.

## Visuals

image
image

## Pre-review Checklist

-   [x] Acceptance criteria satisfied and marked in a related issue
-   [x] Relevant `@since` tags included in DocBlocks
-   [x] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

## Testing Instructions

- "Acceptance Criteria" of the issue must be pass when test this pr.

### CODE REVIEW

File: src/Tracking/UsageTrackingOnBoarding.php [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+		$allowTrackingLink = sprintf(
+			'<br><br><a href="%3$s" class="button-secondary">%1$s</a>&nbsp;&nbsp;<a href="%4$s" class="button-secondary">%2$s</a>',
+			esc_html__( 'Allow', 'give' ),
+			esc_html__( 'Do not allow', 'give' ),
+			add_query_arg( [ 'give_action' => 'opt_in_into_tracking' ] ),
+			add_query_arg( [ 'give_action' => 'opt_out_into_tracking' ] )
+		);
+
+		give()->notices->register_notice(
```
Comment (kjohnson) [inline src/Tracking/UsageTrackingOnBoarding.php:48]: We should specify a notice type other than the default, `error`.

Perhaps `info`?

**Review by kjohnson — APPROVED** [review]
Looks good. I'd like to also consider registering a section in Onboarding and/or Setup, which was discussed during a previous cycle, but scope hammered out.

File: src/Tracking/UsageTrackingOnBoarding.php [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+	 * @sicne 2.10.0
+	 */
+	public function addNotice() {
+		$isAdminOptedIn = give_is_setting_enabled( give_get_option( AdminSettings::USAGE_TRACKING_OPTION_NAME, 'disabled' ) );
+
+		if ( $isAdminOptedIn || ! current_user_can( 'manage_give_settings' ) ) {
+			return;
+		}
+
+		$notice       = esc_html__( 'You can contribute to improve GiveWP. If you opt-in to "Usage Tracking" then we will track non-sensitive data of your website. We will use this information to improve plugin.', 'give' );
```
Comment (kjohnson) [inline src/Tracking/UsageTrackingOnBoarding.php:34]: Also, we'll want @DevinWalker and/or @mathetos to review the public facing message.

File: src/Tracking/UsageTrackingOnBoarding.php (hunk 1 above)
Comment (ravinderk) [reply src/Tracking/UsageTrackingOnBoarding.php:48]: fixed: https://github.com/impress-org/givewp/compare/c999dbabcf00...80ff0aba96c3

File: src/Tracking/UsageTrackingOnBoarding.php (hunk 2 above)
Comment (DevinWalker) [reply src/Tracking/UsageTrackingOnBoarding.php:34]: @ravinderk see the way Gutenberg asks for permissions. Let's format it like that with the first bold sentence, italicized second sentence, and the "Yes" "No" buttons.

> **Can GiveWP collect data about the usage of the plugin?**
> *Usage data is completely anonymous, does not include any personal information, and will only be used to improve the software.*

File: src/Tracking/AdminSettings.php [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+		}
+
+		array_splice(
+			$settings,
+			-2,
+			0,
+			[
+				[
+					'name'    => __( 'Usage Tracking', 'give' ),
+					'label'   => esc_html__( 'Allow usage of GiveWP to be tracked.', 'give' ),
```
Comment (DevinWalker) [inline src/Tracking/AdminSettings.php:45]: Change the label to to "Anonymous Usage Tracking".

File: src/Tracking/AdminSettings.php [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+					'desc'    => sprintf(
+						'%1$s <a href="%2$s">%3$s</a>',
+						esc_html__( 'To opt out, choose "Disabled". Your website remains untracked, and no data will be collected. Read about what usage data is tracked at:', 'give' ),
```
Comment (DevinWalker) [inline src/Tracking/AdminSettings.php:48]: Change to:

> Can GiveWP collect data about the usage of the plugin? Usage data is completely anonymous, does not include any personal information, and will only be used to improve the software.

**Review by DevinWalker — CHANGES_REQUESTED** [review]

**Comment by ravinderk** [comment]
@DevinWalker Changes implemented as per your suggesitons.

image
image

File: src/Tracking/AdminSettings.php (hunk 4 above)
Comment (ravinderk) [reply src/Tracking/AdminSettings.php]: fixed: https://github.com/impress-org/givewp/pull/5460#issuecomment-730881805

File: src/Tracking/AdminSettings.php (hunk 3 above)
Comment (ravinderk) [reply src/Tracking/AdminSettings.php:45]: fixed: https://github.com/impress-org/givewp/pull/5460#issuecomment-730881805

File: src/Tracking/UsageTrackingOnBoarding.php (hunk 2 above)
Comment (ravinderk) [reply src/Tracking/UsageTrackingOnBoarding.php:34]: fixed: https://github.com/impress-org/givewp/pull/5460#issuecomment-730881805

**Pull request merged**